backend/project-management-system/src/main/java/org/egov/works/enrichment/ProjectEnrichment.java

```diff
@@ -0,0 +1,130 @@
+package org.egov.works.enrichment;
+
+import digit.models.coremodels.AuditDetails;
+import digit.models.coremodels.IdResponse;
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.tracer.model.CustomException;
+import org.egov.works.config.ProjectConfiguration;
+import org.egov.works.repository.IdGenRepository;
+import org.egov.works.util.ProjectServiceUtil;
+import org.egov.works.web.models.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+public class ProjectEnrichment {
+
+    @Autowired
+    private ProjectServiceUtil projectServiceUtil;
+
+    @Autowired
+    private IdGenRepository idGenRepository;
+
+    @Autowired
+    private ProjectConfiguration config;
+
+    public void enrichCreateProject(ProjectRequest request) {
+        RequestInfo requestInfo = request.getRequestInfo();
+        List<Project> projects = request.getProjects();
+
+        String rootTenantId = projects.get(0).getTenantId().split("\\.")[0];
+
+        List<String> projectNumbers = getIdList(requestInfo, rootTenantId
+                , config.getIdgenProjectNumberName(), config.getIdgenProjectNumberFormat(), projects.size());
+
+        int i = 0;
+        for (Project project: projects) {
+
+            AuditDetails auditDetails = projectServiceUtil.getAuditDetails(requestInfo.getUserInfo().getUuid(), null, true);
+            project.setAuditDetails(auditDetails);
+            project.setId(UUID.randomUUID().toString());
+
+            if (projectNumbers != null && !projectNumbers.isEmpty()) {
+                projects.get(i).setProjectNumber(projectNumbers.get(i));
+                i++;
+            }
+
+            for (Target target: project.getTargets()) {
+                target.setId(UUID.randomUUID().toString());
+                target.setAuditDetails(auditDetails);
+            }
+
+            for (Document document: project.getDocuments()) {
+                document.setId(UUID.randomUUID().toString());
+                document.setAuditDetails(auditDetails);
+            }
+
+            if (project.getAddress() != null) {
+                project.getAddress().setAuditDetails(auditDetails);
+                project.getAddress().setId(UUID.randomUUID().toString());
+            }
+        }
+
+    }
+
+    public void enrichUpdateProject(ProjectRequest request, List<Project> projectsFromDB) {
```

Comment:
enrichProjectOnUpdate. Please make your method names very readable.

Reply:
Did the corresponding changes in latest push. Method name changed to enrichProjectOnUpdate